test/cereal_test.exs

```diff
@@ -2,7 +2,75 @@ defmodule CerealTest do
   use ExUnit.Case
   doctest Cereal
 
-  test "the truth" do
-    assert 1 + 1 == 2
+  defmodule UserSerializer do
+    use Cereal.Serializer
+    attributes [:name, :property_that_needs_context]
+
+    def property_that_needs_context(_, %{assigns: %{article_name: article_name, comment_id: comment_id}}) do
+      "My context is #{article_name} and comment with id #{comment_id}"
+    end
+  end
+
+  defmodule CommentSerializer do
+    use Cereal.Serializer
+    attributes [:text, :property_that_needs_context]
+    has_one :user, serializer: UserSerializer, include: true
+
+    def property_that_needs_context(_, %{assigns: %{article_name: article_name, request_id: request_id}}) do
+      "My context is #{article_name} and I still have access to the request id #{request_id}"
+    end
+
+    def assigns(%{id: id}, _) do
+      %{comment_id: id}
+    end
+    def assigns(_, conn), do: conn
```

Comment:
A couple small thoughts here:

1. Shouldn't the default return by empty map `%{}`, not the whole conn?
2. I think it might make a little more sense to lean on the overrideable function head in the default case, like:

```
def assigns(%{id: id}, _), do: %{comment_id: id}
def assigns(data, conn), do: super(data, conn)
```